### PR TITLE
Fix script checkout

### PR DESCRIPTION
**Pull request by d4r1091**
## Context

An issue started to appear on CircleCI when the build was referencing the old version number.

## Approach

- Tried to build locally, all good as expected
- Then, came back to analyzing the script.

### Findings

The script that compares the `MARKETING_VERSION` triggers a checkout to the previous Git commit.
The root cause is that the state of the codebase if the check succeded, was kept as per its previous commit compared to the `origin`, hence, the previous number 🤦 .

The reason for not spotting this earlier is the fact that this new approach was tested without applying a `pilot` to send it to the store for real, as the update essentially moved the version attribution from a `VERSION` file to the` xcconfig` one. Still, the detail on the version check played a role here 😉 

This PR fixes it 😇 

## Other

Triggered a job to validate it, finalizing the upload from this branch (on top of validating it locally) and worked 👍 . I've removed the tag already 😉 .

### CODE REVIEW

File: .circleci/deploy-release.yml [hunk 1]
```diff
@@ -33,7 +30,12 @@ jobs:
             if [ "$CURRENT_VERSION" = "$OLD_VERSION" ]; then
               echo "MARKETING_VERSION has not changed. Exiting..."
               circleci-agent step halt
+              exit 0
             fi
+
+            git fetch origin $CIRCLE_BRANCH
```
Comment (lucaschifino) [inline .circleci/deploy-release.yml:36]: Is this line needed? I guess we already have the up-to-date version locally at this point and checking out (as done on the line below) would be enough, wouldn't it?

**Review by lucaschifino — DISMISSED** [review]
Nice catch  ⚽ 🥅

Added a small non-blocking question

File: .circleci/deploy-release.yml (hunk 1 above)
Comment (d4r1091) [reply .circleci/deploy-release.yml]: So happy when I receive such comments 🙏 
Not needed indeed 😄 - thought the same when reviewing the script.
Left there as it was part of the old process. Removed in last commits 😉

File: .circleci/deploy-release.yml [hunk 2]
```diff
@@ -33,7 +30,11 @@ jobs:
             if [ "$CURRENT_VERSION" = "$OLD_VERSION" ]; then
               echo "MARKETING_VERSION has not changed. Exiting..."
               circleci-agent step halt
+              exit 0
             fi
+
+            git checkout $CIRCLE_BRANCH
```
Comment (ecotopian) [inline .circleci/deploy-release.yml:36]: Great catch 🕵️

**Review by ecotopian — APPROVED** [review]
🧹

**Review by lucaschifino — APPROVED** [review]
🟢🕵️🔨

**Pull request merged**